### ISSUE TITLE
[bug] Remote randomly wakes up and show screen if media player is in the activity screen 

### ISSUE THREAD

**Issue by Tobbes1990** [unfoldedcircle/bug-reports#409]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When i set up the media player on the active activity screen (where you can see the picture of the current playing Movie and the remaining time) in the Apple TV or kodi Integration the remote randomly wakes up and screen goes on.

### How to Reproduce

Start an activity with media player widget on screen
Lay down the remote while playing
Remotes wakes up randomly 

### Expected behavior

Screen should turned off unless you press a button or take it in hands

### System version

2.0

### What part of the system affected by the problem?

UI

**Comment by kennymc-c** [comment]
Have you checked if there is any device in the same network as the remote that sends wake on lan magic packets to the remote? If the display activates by itself the remote was usually woken up by WOL.

**Comment by Tobbes1990** [comment]
I dont think so. It only happens when i play a movie (on Apple TV or kodi, it doesn’t matter).
I have homeassistant with the unfolded circle integration, is it possible that this integration sends the magic packet?

**Comment by kennymc-c** [comment]
At least the latest pre-Release version supports WoL. But the author said that it should only wake up the remote if the user performs an action within HA and it can also be deactivated.

**Comment by Tobbes1990** [comment]
I deactivated the integration in HA and the problem is still there. I have no other device in my network that sends wake on lan packets. Any other idea?

**Comment by kennymc-c** [comment]
You could try to put the remote into a separate network without any other devices and wait if the display still activates by itself. Can you see core log entries that mention something like wake on lan when this happens?
Have you tried to deactivate Wake on lan on the remote?

**Comment by RC-Thoughts** [comment]
I am investigating the same issue with 2.0.0. I can confirm there are no WoL-packets going around in network. I have traced the biggest suspect ie. HA straight from source and also network, no WoL packets floating around or being broadcasted. I have equipment for wireless trace on packet level so I may need to take that too to try to see if there is any correlation in wifi-packets to screen turning on.

The difference to OP in my case is that this happens in all activities regardless if they have a media player on screen or not. Screen lights up ~every two minutes, sometimes it can be up to 6 minutes.

**Comment by Tobbes1990** [comment]
How do I deactivate WOL on the remote? I can't find it in settings on the web configurator

**Comment by Tobbes1990** [comment]
> I am investigating the same issue with 2.0.0. I can confirm there are no WoL-packets going around in network. I have traced the biggest suspect ie. HA straight from source and also network, no WoL packets floating around or being broadcasted. I have equipment for wireless trace on packet level so I may need to take that too to try to see if there is any correlation in wifi-packets to screen turning on.
> 
> The difference to OP in my case is that this happens in all activities regardless if they have a media player on screen or not. Screen lights up ~every two minutes, sometimes it can be up to 6 minutes.

You're right. It happens in all activities.

**Comment by kennymc-c** [comment]
> How do I deactivate WOL on the remote? I can't find it in settings on the web configurator

You need to do it on the remote itself

**Comment by Tobbes1990** [comment]
Am I dumb? I can't find this option... Or do you mean the new option to always stay connected to the wifi (since version 2.0.0)? If so, this option is a dealbreaker for me and my use case.

**Comment by kennymc-c** [comment]
It's the option you mentioned. Just test it without it enabled and if the display still activates you can rule out WoL as a reason for this issue

**Comment by RC-Thoughts** [comment]
I have confirmed the waking up is done by R2, not by anything in network. I took a wifi-capture of all packets in connected SSID and I can clearly see that when R2 is sleeping it does the usual PS-Poll and ACK etc, no issues. 

At the same time R2 wakes up on it's own first R2 sends  a NULL data to Access point (AP),  after that is ACKed R2 sends a Ready to send (RTS) packet. This basically tells AP that R2 want's to send something. 

I will take a longer capture and try to identify any pattern. There's also other findings in there but that's for another topic.

**Comment by RC-Thoughts** [comment]
Can also confirm that disabling new reconnection logic also stops this "waking up"-behavior.

**Comment by WillTriumph1** [comment]
So logically we would want the remote “awake” to rapidly connect.  So is possible to wake up/stay awake w/o the screen coming on unless triggered as usual (removal from dock, movement sensitive, etc.)

**Comment by Tobbes1990** [comment]
> So logically we would want the remote “awake” to rapidly connect. So is possible to wake up/stay awake w/o the screen coming on unless triggered as usual (removal from dock, movement sensitive, etc.)

That's the behaviour it should be. 
I also had disabled the new logic. The problems were gone. 
I think we have to wait for a fix from the developers.

**Comment by RC-Thoughts** [comment]
As information, R2 logs and WLAN-traces delivered to Unfolded, let's see what they come up with.

**Comment by RC-Thoughts** [comment]
Any progress on log-investigation?

**Comment by Tobbes1990** [comment]
Now, my R2 wakes up no matter whether the remote is in use or not. Even disabling the new logic has no effect.

**Comment by Tobbes1990** [comment]
This is the log:
2024-11-05 20:18:20.386087 +00:00	core	NOTICE	[integrations] WiFi connected: starting all external intg drivers
2024-11-05 20:18:20.384644 +00:00	core	NOTICE	WiFi interface is up and routable, sending connected event
2024-11-05 20:18:20.383178 +00:00	core	NOTICE	Network changed: link_name: wlan0, operational_state: routable, address_state: routable, ipv4_address_state: routable
2024-11-05 20:18:20.326827 +00:00	core	NOTICE	Network changed: link_name: wlan0, operational_state: degraded, address_state: degraded, ipv6_address_state: degraded
2024-11-05 20:18:20.303130 +00:00	core	WARN	Transition SetNormalMode not allowed in state Running
2024-11-05 20:18:20.291434 +00:00	core	NOTICE	Setting button backlight to: 50%


At this point the r2 wakes up. This happens every 10 minutes (20:28:20, 20:38:20...)

**Comment by martonborzak** [comment]
This is most likely an issue with the chip firmware itself. We're in dialog with the manufacturer, but the firmware itself is a closed source binary, so nothing we can do until the manufacturer fixes the issue.

**Comment by Tobbes1990** [comment]
Thanks for the replay. This issue first happens on beta firmware 2.0.0. On the last stable firmware, everything was alright. But I love the custom integrations so i don't want to go back :)

**Comment by martonborzak** [comment]
We've updated the WiFi chip firmware we got from the vendor in that beta release, that's why. We're on it, but it's a waiting game until we hear back from them.

**Comment by Tobbes1990** [comment]
New Beta Version - the Problem still exists :(

**Comment by martonborzak** [comment]
Please read the release notes :) What is mentioned there included in the update.

We still haven't received a working firmware from the vendor. As soon as we do, we'll of course let you know here and will push an update as well.

**Comment by RC-Thoughts** [comment]
For whatever reason this issue is not appearing in 2.2.1 beta for me.

**Comment by Tobbes1990** [comment]
Nothing changed for me. But there are also no fixes in the changelog. So i have to wait.

**Comment by Tobbes1990** [comment]
Hey Marton,
The bug has been here in the forum for at least 3 months now. When can you finally expect a fix?

**Comment by Tobbes1990** [comment]
Another update - nothing new. 4 month. Really ?

**Comment by Originalme** [comment]
I have this, and can reproduce it constantly. I saw others mention a way of pulling logs from the device, is there a KB or guide on that I can reference?

**Comment by kennymc-c** [comment]
@Originalme How do you reproduce and trigger the screen wake-up every time? So far all reports indicate it's completly random if keep Wifi active during standby is enabled.
You can download log under settings/development/logs

**Comment by Originalme** [comment]
Sorry for the delay, I was traveling for work. I will grab the logs next time I am around it. It happens constantly, probably every 1~5 minutes. I can repro this easily enough.

**Comment by kennymc-c** [comment]
It could also be another device in your network or your router causing wake ups. To be sure the remote should be the only device in the network when you do a repro.

**Comment by zehnm** [comment]
The just released beta firmware 2.9.8 contains a fix for repeated wakeup (every few minutes) if the "keep WiFi connected in standby" feature is enabled. The device should now only wake if required for Wi‑Fi security updates (for example, WPA key rotation / rekeying), which is usually every few hours (depending on access point configuration).
I say "should" because this is not a traditional bug with an easy fix. This happens inside the closed proprietary WiFi controller manufacturer firmware, that signals the system to wake up because it needs the host wifi driver for a specific action.

Please report back if that solves your issue.
We are also working on a public release, but still have a few more issues to fix.

**Comment by sjftech** [comment]
These newer firmwares since 2.9.8 alongside forcing 2.4GHz on the remote seems to fix this issue for me with a Unifi network. Strangely, 5GHz still randomly wakes up for me.

**Comment by zehnm** [comment]
We have found an issue in the 5 GHz WiFi controller roaming logic for sleep and wakeup that will be fixed in the next update.

**Comment by sjftech** [comment]
Upgraded to 2.9.13 and and 5 GHz still causes the screen to wake. Switch back to 2.4 GHz and the remote is sleeping again. This is the exact same SSID and access point, just changing the setting on the remote itself.
